public_html/wp-content/plugins/wordcamp-coming-soon-page/classes/wordcamp-coming-soon-page.php

```diff
@@ -293,15 +297,31 @@ public function get_contact_form_shortcode() {
 
 					ob_start();
 					echo do_shortcode( $shortcode[0] );
-					$contact_form_shortcode = ob_get_clean();
+					$contact_form_content = ob_get_clean();
 
 					wp_reset_postdata();
 					break;
 				}
 			}
+
+			// Along with shortcodes, also check for blocks.
+			if ( ! has_blocks( $page->post_content ) || $contact_form_content ) {
+				continue;
+			}
+			$blocks = parse_blocks( $page->post_content );
+			foreach( $blocks as $block ) {
+				if ( 'jetpack/contact-form' === $block['blockName'] ) {
+					global $post;
+					$post = $page;
+					setup_postdata( $post );
+					$contact_form_content = render_block( $block );
+					wp_reset_postdata();
+					break;
+				}
+			}
```

Comment:
It might be a bit more readable to structure it like this:

```php
foreach ( $all_pages as $page ) {
	if ( has_block( 'jetpack/contact-form' ) {
		// get content from block
	} elseif ( has_shortcode( 'contact_form' ) {
		// get content from shortcode
	}

	if ( $contact_form_content ) {
		break;
	}
}
```

I think the original code may have been written before `has_shortcode()` existed, so it'd be nice to update it to take advantage of the function, since it'll make the structure of the code more obvious and descriptive.

Placing the `break` after checking the block/shortcode prevents an extra unnecessary iteration of the loop. That'd be a useless micro-optimization from a performance perspective, but it feels a bit more logical to me than, since it happens right after the condition might be met.